libraries/AP_FETtecOneWire/README.md

```diff
@@ -0,0 +1,168 @@
+- Optionaly dynamically change the center frequency of the notch filters used to reduce frame vibration noise in the gyros
+- Optionaly measure battery voltage and power consumption
+
+
+## Full/Alternative Telemetry
+The telemetry can be switched to "per ESC" Mode, where one ESC answers with it's full telemetry as oneWire package including CRC and additionally the CRC Erros counted by the ESC..
+To use this mode OW_SET_TLM_TYPE must be send to each ESC. It make sense to set it while initializing.
+If this was successful set the ESC response with "OW OK".
+
+The answer is packed inside a OW package, that can be received with the FETtecOneWire::receive function, that also checks the CRC.
+
+As the packages are send in an uInt8_t array the values must be restored like as only temp is one byte long:
+        Telemetry[0]= telem[0]; //Temp
+        Telemetry[1]=(telem[1]<<8)|telem[2];//Volt
+        Telemetry[2]=(telem[3]<<8)|telem[4];//Current
+        Telemetry[3]=(telem[5]<<8)|telem[6];//ERPM
+        Telemetry[4]=(telem[7]<<8)|telem[8];//Consumption
+        Telemetry[5]=(telem[9]<<8)|telem[10];//CRCerr
+
+
+
+
+## Extra features
+
+The ESC can beep and have lights. To control this you must add this code snippet to the header file:
+
+```C++
+public:
+/**
+    makes all connected ESCs beep
+    @param beepFrequency a 8 bit value from 0-255. higher make a higher beep
+*/
+    void Beep(uint8_t beepFrequency);
+
+/**
+    sets the racewire color for all ESCs
+    R, G, B = 8bit colors
+*/
+    void RW_LEDcolor(uint8_t R, uint8_t G, uint8_t B);
+```
+
+And this code snippet to the .cpp file:
+
+```C++
+/**
+    makes all connected ESCs beep
+    @param beepFrequency a 8 bit value from 0-255. higher make a higher beep
+*/
+void AP_FETtecOneWire::Beep(uint8_t beepFrequency)
+{
+    if (_id_count > 0) {
+        const uint8_t request[2] = {OW_BEEP, beepFrequency};
+        const uint8_t request_len[1] = {2};
+        const uint8_t spacer[2] = {0, 0};
+        for (uint8_t i = _min_id; i < _max_id + 1; i++) {
+            Transmit(i, request, request_len);
+            // add two zeros to make sure all ESCs can catch their command as we don't wait for a response here
+            _uart->write(spacer, 2);
+        }
+    }
+}
+
+/**
+    sets the racewire color for all ESCs
+    @param R red brightness
+    @param G green brightness
+    @param B blue brightness
+*/
+void AP_FETtecOneWire::RW_LEDcolor(uint8_t R, uint8_t G, uint8_t B)
+{
+    if (_id_count > 0) {
+        const uint8_t request[4] = {OW_SET_LED_TMP_COLOR, R, G, B};
+        const uint8_t request_len[1] = {4};
+        const uint8_t spacer[2] = {0, 0};
+        for (uint8_t i = _min_id; i < _max_id + 1; i++) {
+            Transmit(i, request, request_len);
+            // add two zeros to make sure all ESCs can catch their command as we don't wait for a response here
+            _uart->write(spacer, 2);
+        }
+    }
+}
```

Comment:
Let's include the cool stuff, but behind additional `#if`s - on by default

Reply:
done